sql-plugin/src/main/330+/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala

```diff
@@ -184,6 +186,113 @@ object IntervalUtils extends Arm {
       z.castTo(DType.INT64)
     }
   }
+
+  def getDouble(s: Scalar): Double = {
+    s.getType match {
+      case DType.INT8 => s.getByte.toDouble
+      case DType.INT16 => s.getShort.toDouble
+      case DType.INT32 => s.getInt.toDouble
+      case DType.INT64 => s.getLong.toDouble
+      case DType.FLOAT32 => s.getFloat.toDouble
+      case DType.FLOAT64 => s.getDouble
+      case _ => throw new IllegalArgumentException()
```

Comment:
While we don't expect to ever hit this, if we do then it will be annoying that the exception didn't bother to state what the unexpected type was.
```suggestion
      case t =>
        throw new IllegalArgumentException(s"Unexpected type $t")
```
Comment applies to the other instances of this below.
Comment applies to the

Reply:
done